app/views/supermarkets/show.html.erb

```diff
@@ -0,0 +1,8 @@
+<h1><%= @supermarket.name %></h1>
+
+<p>
+  Customers that have shopped here: 
+  <% @supermarket.all_customers.each do |customer| %>
+    <%= customer == @supermarket.all_customers.last ? "#{customer}" : "#{customer}, " %> 
```

Comment:
I don't think you need to do this much in here. You are already iterating over the `all_customers` method that you created. And since you used `.pluck` for the names, you just need to return the `customer`.